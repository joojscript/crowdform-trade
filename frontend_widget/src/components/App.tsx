import { useEffect, useState } from "react";

import { SupportedChainId, SwapWidget } from "@uniswap/widgets";

import "@uniswap/widgets/fonts.css";

import { useConnectWallet } from "@web3-onboard/react";
import { ethers } from "ethers";

import { TOKEN_LIST } from "../constants";
import styles from "../styles/Home.module.css";

const USDC = TOKEN_LIST[0].address; // USDC contract address for Base Network

// Hexadecimal representation of chainId for BASE Mainnet:
const BASE_CHAIN_ID = SupportedChainId.BASE.toString(
  16
) as unknown as SupportedChainId;

export default function App() {
  const [{ wallet, connecting }, connect, disconnect] = useConnectWallet();

  const [provider, setProvider] = useState<ethers.providers.Web3Provider>();

  useEffect(() => {
    if (wallet?.provider) {
      setProvider(new ethers.providers.Web3Provider(wallet.provider, "any"));
    } else {
      setProvider(undefined);
    }
  }, [wallet]);

  // The connect wallet function which will be based to the Uniswap component below.
  const connectWallet = () => {
    connect();
  };

  return (
    <div className={styles.container}>
      <main className={styles.main}>
        <h1 className={styles.title}>Uniswap Swap Widget</h1>

        <div className={styles.demo}>
          <div className={styles.widget}>
            <SwapWidget
              data-testid="swap-widget"
              defaultChainId={BASE_CHAIN_ID}
              tokenList={TOKEN_LIST}
              provider={provider}
              // When the Uniswap connect wallet button gets hit
              // the function gets called. We'll hook this up to
              // our connect wallet method from web3-onboard.
              onConnectWalletClick={connectWallet}
              defaultInputTokenAddress="NATIVE"
              defaultInputAmount="1"
              defaultOutputTokenAddress={USDC}
            />
          </div>
        </div>
      </main>
    </div>
  );
}
