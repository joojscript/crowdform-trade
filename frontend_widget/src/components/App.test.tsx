import { render, waitFor } from "@testing-library/react";
import React from "react";
import { describe, it, vi } from "vitest";
import App from "./App";

vi.mock("@web3-onboard/react", () => ({
  useConnectWallet: vi.fn().mockReturnValue([
    {
      wallet: {},
      connecting: true,
    },
    vi.fn(),
    vi.fn(),
  ]),
}));

describe("<App />", () => {
  it("should match snapshot", async () => {
    const { container } = await render(<App />);

    expect(container).toMatchSnapshot();
  });

  it("should correctly instantiate the provider", async () => {
    const { container } = await render(<App />);
    const useStateSpy = vi.spyOn(React, "useState");
    // eslint-disable-next-line testing-library/no-container, testing-library/no-node-access
    const swapWidget = container.querySelector(
      'div[class~="WidgetWrapper__StyledWidgetWrapper"]'
    );

    await waitFor(() => {
      expect(useStateSpy).toHaveBeenCalled();
    });
    expect(swapWidget).toBeDefined();
  });
});
