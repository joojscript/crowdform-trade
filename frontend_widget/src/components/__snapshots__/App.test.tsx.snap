// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<App /> > should be true 1`] = `
<div>
  <div
    class="container"
  >
    <main
      class="main"
    >
      <h1
        class="title"
      >
        Uniswap Swap Widget
      </h1>
      <div
        class="demo"
      >
        <div
          class="widget"
        >
          <div
            class="WidgetWrapper__StyledWidgetWrapper-sc-11krezp-0 eiYzsD"
            width="360"
          />
        </div>
      </div>
    </main>
  </div>
</div>
`;

exports[`<App /> > should match snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <main
      class="main"
    >
      <h1
        class="title"
      >
        Uniswap Swap Widget
      </h1>
      <div
        class="demo"
      >
        <div
          class="widget"
        >
          <div
            class="WidgetWrapper__StyledWidgetWrapper-sc-11krezp-0 eiYzsD"
            width="360"
          />
        </div>
      </div>
    </main>
  </div>
</div>
`;
